### ISSUE TITLE
Release 0.4

### ISSUE THREAD

**Issue by ldaley** [ReactiveX/RxJavaReactiveStreams#11]
I've made some refinements, and AFAIK the implementation is now at RC status. At this early stage, I think we should just go ahead and roll out 0.4.

Before the next release, I'd like to get some more test coverage into the build by way of tests that actually exercise interop. At the moment we only have that with Ratpack.

@benjchristensen note that we've gone back to a multi project build now, so we may have to work through those issues.

**Comment by benjchristensen** [comment]
Trying to release gets errors as it's not correctly creating the artifacts:

```
rxjava-reactive-streams:check
:rxjava-reactive-streams:compileExamplesJava UP-TO-DATE
:rxjava-reactive-streams:processExamplesResources UP-TO-DATE
:rxjava-reactive-streams:examplesClasses UP-TO-DATE
:rxjava-reactive-streams:build
:build
:verifySnapshotStatus UP-TO-DATE
:rxjava-reactive-streams:verifySnapshotStatus UP-TO-DATE
:rxjava-reactive-streams:artifactoryPublish
:artifactoryPublish FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':artifactoryPublish'.
> File '/scratch/jenkins/workspace/ReactiveX/RxJavaReactiveStreams/RxJavaReactiveStreams-0.x-release/rxjava-reactive-streams/build/libs/rxjava-reactive-streams.jar.asc' does not exists, and need to be published from publication mavenNebula

* Try:
Run with --info or --debug option to get more log output.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':artifactoryPublish'.
    at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:69)
    at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:46)
    at org.gradle.api.internal.tasks.execution.PostExecutionAnalysisTaskExecuter.execute(PostExecutionAnalysisTaskExecuter.java:35)
    at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:64)
    at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
    at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:42)
    at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
    at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:53)
    at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
    at org.gradle.api.internal.AbstractTask.executeWithoutThrowingTaskFailure(AbstractTask.java:289)
    at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.executeTask(AbstractTaskPlanExecutor.java:79)
    at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.processTask(AbstractTaskPlanExecutor.java:63)
    at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.run(AbstractTaskPlanExecutor.java:51)
    at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor.process(DefaultTaskPlanExecutor.java:23)
    at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter.execute(DefaultTaskGraphExecuter.java:86)
    at org.gradle.execution.SelectedTaskExecutionAction.execute(SelectedTaskExecutionAction.java:29)
    at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:61)
    at org.gradle.execution.DefaultBuildExecuter.access$200(DefaultBuildExecuter.java:23)
    at org.gradle.execution.DefaultBuildExecuter$2.proceed(DefaultBuildExecuter.java:67)
    at org.gradle.execution.DryRunBuildExecutionAction.execute(DryRunBuildExecutionAction.java:32)
    at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:61)
    at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:54)
    at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:166)
    at org.gradle.initialization.DefaultGradleLauncher.doBuild(DefaultGradleLauncher.java:113)
    at org.gradle.initialization.DefaultGradleLauncher.run(DefaultGradleLauncher.java:81)
    at org.gradle.launcher.exec.InProcessBuildActionExecuter$DefaultBuildController.run(InProcessBuildActionExecuter.java:64)
    at org.gradle.launcher.cli.ExecuteBuildAction.run(ExecuteBuildAction.java:33)
    at org.gradle.launcher.cli.ExecuteBuildAction.run(ExecuteBuildAction.java:24)
    at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:35)
    at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:26)
    at org.gradle.launcher.cli.RunBuildAction.run(RunBuildAction.java:50)
    at org.gradle.internal.Actions$RunnableActionAdapter.execute(Actions.java:171)
    at org.gradle.launcher.cli.CommandLineActionFactory$ParseAndBuildAction.execute(CommandLineActionFactory.java:201)
    at org.gradle.launcher.cli.CommandLineActionFactory$ParseAndBuildAction.execute(CommandLineActionFactory.java:174)
    at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:170)
    at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:139)
    at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:33)
    at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:22)
    at org.gradle.launcher.Main.doAction(Main.java:46)
    at org.gradle.launcher.bootstrap.EntryPoint.run(EntryPoint.java:45)
    at org.gradle.launcher.Main.main(Main.java:37)
    at org.gradle.launcher.bootstrap.ProcessBootstrap.runNoExit(ProcessBootstrap.java:50)
    at org.gradle.launcher.bootstrap.ProcessBootstrap.run(ProcessBootstrap.java:32)
    at org.gradle.launcher.GradleMain.main(GradleMain.java:23)
    at org.gradle.wrapper.BootstrapMainStarter.start(BootstrapMainStarter.java:30)
    at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:127)
    at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:55)
Caused by: org.gradle.api.GradleException: File '/scratch/jenkins/workspace/ReactiveX/RxJavaReactiveStreams/RxJavaReactiveStreams-0.x-release/rxjava-reactive-streams/build/libs/rxjava-reactive-streams.jar.asc' does not exists, and need to be published from publication mavenNebula
    at org.jfrog.gradle.plugin.artifactory.task.BuildInfoPublicationsTask.createBuilder(BuildInfoPublicationsTask.java:245)
    at org.jfrog.gradle.plugin.artifactory.task.BuildInfoPublicationsTask.getArtifactDeployDetails(BuildInfoPublicationsTask.java:228)
    at org.jfrog.gradle.plugin.artifactory.task.BuildInfoPublicationsTask_Decorated.getArtifactDeployDetails(Unknown Source)
    at org.jfrog.gradle.plugin.artifactory.task.BuildInfoPublicationsTask.collectDescriptorsAndArtifactsForUpload(BuildInfoPublicationsTask.java:155)
    at org.jfrog.gradle.plugin.artifactory.task.BuildInfoBaseTask.prepareAndDeploy(BuildInfoBaseTask.java:267)
    at org.jfrog.gradle.plugin.artifactory.task.BuildInfoBaseTask.collectProjectBuildInfo(BuildInfoBaseTask.java:365)
    at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:63)
    at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.doExecute(AnnotationProcessingTaskFactory.java:219)
    at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:212)
    at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:201)
    at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:533)
    at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:516)
    at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:80)
    at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:61)
    ... 46 more


BUILD FAILED
```

**Comment by benjchristensen** [comment]
While getting this working do you want to help fix the Nebula plugins to achieve this? -> https://github.com/ReactiveX/RxJava/issues/1810

**Comment by ldaley** [comment]
@benjchristensen just changed some config, try again please.

**Comment by benjchristensen** [comment]
Errors:

```
:rxjava-reactive-streams:verifyReleaseStatus UP-TO-DATE
:rxjava-reactive-streams:bintrayUploadSkipping upload for missing file '/scratch/jenkins/workspace/ReactiveX/RxJavaReactiveStreams/RxJavaReactiveStreams-0.x-release/rxjava-reactive-streams/build/libs/rxjava-reactive-streams.jar.asc'.
Skipping upload for missing file '/scratch/jenkins/workspace/ReactiveX/RxJavaReactiveStreams/RxJavaReactiveStreams-0.x-release/rxjava-reactive-streams/build/libs/rxjava-reactive-streams-javadoc.jar.asc'.
Skipping upload for missing file '/scratch/jenkins/workspace/ReactiveX/RxJavaReactiveStreams/RxJavaReactiveStreams-0.x-release/rxjava-reactive-streams/build/libs/rxjava-reactive-streams-sources.jar.asc'.
Skipping upload for missing file '/scratch/jenkins/workspace/ReactiveX/RxJavaReactiveStreams/RxJavaReactiveStreams-0.x-release/rxjava-reactive-streams/build/libs/rxjava-reactive-streams-benchmarks.jar.asc'.
Skipping upload for missing file '/scratch/jenkins/workspace/ReactiveX/RxJavaReactiveStreams/RxJavaReactiveStreams-0.x-release/rxjava-reactive-streams/build/poms/mavenNebula.pom.asc'.

Package Sync: /maven_central_sync/reactivex/RxJava/RxJavaReactiveStreams/versions/unspecified
null
null
null
Unable to sync to Maven Central, please do manually at https://bintray.com/reactivex/RxJava/rxjava-reactive-streams/unspecified/view/central
:bintrayUpload FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':bintrayUpload'.
> Extension of type 'ScmInfoExtension' does not exist. Currently registered extension types: [DefaultExtraPropertiesExtension, DefaultArtifactPublicationSet_Decorated, ContactsExtension_Decorated, EclipseModel_Decorated, IdeaModel_Decorated, GrgitReleasePluginExtension_Decorated, BintrayExtension_Decorated, DefaultPublishingExtension_Decorated, DependencyLockCommitExtension_Decorated]

* Try:
Run with --info or --debug option to get more log output.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':bintrayUpload'.
    at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:69)
    at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:46)
    at org.gradle.api.internal.tasks.execution.PostExecutionAnalysisTaskExecuter.execute(PostExecutionAnalysisTaskExecuter.java:35)
    at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:64)
    at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
    at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:42)
    at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
    at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:53)
    at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
    at org.gradle.api.internal.AbstractTask.executeWithoutThrowingTaskFailure(AbstractTask.java:289)
    at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.executeTask(AbstractTaskPlanExecutor.java:79)
    at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.processTask(AbstractTaskPlanExecutor.java:63)
    at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.run(AbstractTaskPlanExecutor.java:51)
    at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor.process(DefaultTaskPlanExecutor.java:23)
    at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter.execute(DefaultTaskGraphExecuter.java:86)
    at org.gradle.execution.SelectedTaskExecutionAction.execute(SelectedTaskExecutionAction.java:29)
    at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:61)
    at org.gradle.execution.DefaultBuildExecuter.access$200(DefaultBuildExecuter.java:23)
    at org.gradle.execution.DefaultBuildExecuter$2.proceed(DefaultBuildExecuter.java:67)
    at org.gradle.execution.DryRunBuildExecutionAction.execute(DryRunBuildExecutionAction.java:32)
    at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:61)
    at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:54)
    at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:166)
    at org.gradle.initialization.DefaultGradleLauncher.doBuild(DefaultGradleLauncher.java:113)
    at org.gradle.initialization.DefaultGradleLauncher.run(DefaultGradleLauncher.java:81)
    at org.gradle.launcher.exec.InProcessBuildActionExecuter$DefaultBuildController.run(InProcessBuildActionExecuter.java:64)
    at org.gradle.launcher.cli.ExecuteBuildAction.run(ExecuteBuildAction.java:33)
    at org.gradle.launcher.cli.ExecuteBuildAction.run(ExecuteBuildAction.java:24)
    at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:35)
    at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:26)
    at org.gradle.launcher.cli.RunBuildAction.run(RunBuildAction.java:50)
    at org.gradle.internal.Actions$RunnableActionAdapter.execute(Actions.java:171)
    at org.gradle.launcher.cli.CommandLineActionFactory$ParseAndBuildAction.execute(CommandLineActionFactory.java:201)
    at org.gradle.launcher.cli.CommandLineActionFactory$ParseAndBuildAction.execute(CommandLineActionFactory.java:174)
    at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:170)
    at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:139)
    at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:33)
    at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:22)
    at org.gradle.launcher.Main.doAction(Main.java:46)
    at org.gradle.launcher.bootstrap.EntryPoint.run(EntryPoint.java:45)
    at org.gradle.launcher.Main.main(Main.java:37)
    at org.gradle.launcher.bootstrap.ProcessBootstrap.runNoExit(ProcessBootstrap.java:50)
    at org.gradle.launcher.bootstrap.ProcessBootstrap.run(ProcessBootstrap.java:32)
    at org.gradle.launcher.GradleMain.main(GradleMain.java:23)
    at org.gradle.wrapper.BootstrapMainStarter.start(BootstrapMainStarter.java:30)
    at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:127)
    at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:55)
Caused by: org.gradle.api.UnknownDomainObjectException: Extension of type 'ScmInfoExtension' does not exist. Currently registered extension types: [DefaultExtraPropertiesExtension, DefaultArtifactPublicationSet_Decorated, ContactsExtension_Decorated, EclipseModel_Decorated, IdeaModel_Decorated, GrgitReleasePluginExtension_Decorated, BintrayExtension_Decorated, DefaultPublishingExtension_Decorated, DependencyLockCommitExtension_Decorated]
    at org.gradle.api.internal.plugins.ExtensionsStorage.getHolderByType(ExtensionsStorage.java:99)
    at org.gradle.api.internal.plugins.ExtensionsStorage.getByType(ExtensionsStorage.java:77)
    at org.gradle.api.internal.plugins.DefaultConvention.getByType(DefaultConvention.java:122)
    at org.gradle.api.plugins.ExtensionContainer$getByType.call(Unknown Source)
    at nebula.plugin.rxjavaproject.RxJavaPublishingPlugin$_apply_closure3.doCall(RxJavaPublishingPlugin.groovy:55)
    at org.gradle.api.internal.AbstractTask$ClosureTaskAction.execute(AbstractTask.java:502)
    at org.gradle.api.internal.AbstractTask$ClosureTaskAction.execute(AbstractTask.java:483)
    at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:80)
    at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:61)
    ... 46 more
```

**Comment by ldaley** [comment]
Pushed changes. Try again please.

**Comment by benjchristensen** [comment]
It built and uploaded, but they don't have versions and are `unspecified`: https://bintray.com/reactivex/RxJava/RxJavaReactiveStreams/view/files/io/reactivex/rxjava-reactive-streams

![screen shot 2014-11-10 at 10 16 29 pm](https://cloud.githubusercontent.com/assets/813492/4989089/42b24662-6927-11e4-9b19-2f2b0d28a069.png)

![screen shot 2014-11-10 at 10 16 35 pm](https://cloud.githubusercontent.com/assets/813492/4989091/47e66ee2-6927-11e4-89e9-9a29f73e2dc0.png)

**Comment by ldaley** [comment]
Discard those please.

**Comment by ldaley** [comment]
Should delete the tag too.

**Comment by ldaley** [comment]
Again please.

**Comment by benjchristensen** [comment]
It built and uploaded 0.5.0 but there are problems with the pom files it seems ... it won't sync to Maven Central:

```
Last Sync Errors: Missing Signature: '/io/reactivex/rxjava-reactive-streams/0.5.0/rxjava-reactive-streams-0.5.0-sources.jar.asc' does not exist for 'rxjava-reactive-streams-0.5.0-sources.jar'.
Missing Signature: '/io/reactivex/rxjava-reactive-streams/0.5.0/rxjava-reactive-streams-0.5.0.jar.asc' does not exist for 'rxjava-reactive-streams-0.5.0.jar'.
Missing Signature: '/io/reactivex/rxjava-reactive-streams/0.5.0/rxjava-reactive-streams-0.5.0-benchmarks.jar.asc' does not exist for 'rxjava-reactive-streams-0.5.0-benchmarks.jar'.
Missing Signature: '/io/reactivex/rxjava-reactive-streams/0.5.0/rxjava-reactive-streams-0.5.0.pom.asc' does not exist for 'rxjava-reactive-streams-0.5.0.pom'.
Missing Signature: '/io/reactivex/rxjava-reactive-streams/0.5.0/rxjava-reactive-streams-0.5.0-javadoc.jar.asc' does not exist for 'rxjava-reactive-streams-0.5.0-javadoc.jar'.
Dropping existing partial staging repository.
```

**Comment by benjchristensen** [comment]
Here are the files: https://bintray.com/reactivex/RxJava/RxJavaReactiveStreams/view/files/io/reactivex/rxjava-reactive-streams/0.5.0

**Comment by ldaley** [comment]
It's missing the signatures. Will look into it. May be a day or two.

**Comment by gabrielruiu** [comment]
Maybe issue can be closed, AFAIK version 1.0.1 is the newest one at the moment
